swan-lake/development-tutorials/observability/overview-of-ballerina-observability.md

```diff
@@ -0,0 +1,204 @@
+---
+
+Ballerina Observability is a built-in feature designed to facilitate monitoring and debugging Ballerina applications by collecting, processing, and analyzing performance and operational data. It integrates seamlessly with observability tools and supports three primary pillars of observability: metrics, tracing, and logging.
+
+1. **Metrics** - Provide quantitative insights into the performance of an application, such as request counts, latencies, and error counts.
+2. **Tracing** - Used to track requests/transactions as they flow through different services in a distributed system from the point of entry to exit.
+3. **Logging** - Text records of activities that occurred with relevant information along with the timestamp with various log levels for further analysis.
+
+Observability is understanding a system's state based on the data it generates, such as logs, metrics, and traces. Observability tools and platforms enable developers to monitor, troubleshoot, and optimize their applications effectively.
+
+Ballerina services and any client connectors are observable by default. HTTP/HTTPS and SQL client connectors use semantic tags to make tracing and metrics monitoring more informative.
+
+## Provide observability in Ballerina
+
+This guide demonstrates how to enable Ballerina observability to observe Ballerina services using different tools and platforms. We will use the example given below.
+
+### Example: A simple Ballerina service to observe
+Create a new Ballerina project using the `bal new` command.
+```
+$ bal new observability_demo
+
+Created new package 'observability_demo' at observability_demo.
+```
+
+Delete everything inside the `main.bal` file which is located inside the newly created `observability_demo` directory, and replace it with the following code to create a Ballerina service.
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+
+// Define data structures.
+type Product record {|
+    int id;
+    string name;
+    float price;
+|};
+
+type OrderRequest record {|
+    int productId;
+    int quantity;
+|};
+
+type Order record {|
+    int orderId;
+    int productId;
+    int quantity;
+    float totalPrice;
+|};
+
+// Sample data
+map<Product> products = {
+    "1": {id: 1, name: "Laptop", price: 1200.00},
+    "2": {id: 2, name: "Smartphone", price: 800.00},
+    "3": {id: 3, name: "Headphones", price: 150.00}
+};
+
+map<Order> orders = {};
+int orderCount = 0;
+
+@display {
+    label: "Shopping Service"
+}
+service /shop on new http:Listener(8090) {
+
+    // List available products.
+    resource function get products() returns Product[] {
+        log:printInfo("Fetching product list");
+        return products.toArray();
+    }
+
+    // Add a new product.
+    resource function post product(@http:Payload Product product) returns http:Created|http:Conflict|error? {
```

Comment:
Why do we need `@http:Payload`? @TharmiganK Can you please help with reviewing the sample code?

Reply:
Checked with @TharmiganK and addressed his suggestions